src/consumer/topic.rs

```diff
@@ -95,45 +96,91 @@ impl<T: DeserializeMessage, Exe: Executor> TopicConsumer<T, Exe> {
                     }
                     break;
                 }
-                Err(ConnectionError::PulsarError(
-                    Some(proto::ServerError::ServiceNotReady),
-                    text,
-                )) => {
-                    if operation_retry_options.max_retries.is_none()
-                        || operation_retry_options.max_retries.unwrap() > current_retries
-                    {
-                        error!("subscribe({}) answered ServiceNotReady, retrying request after {}ms (max_retries = {:?}): {}",
-                        topic, operation_retry_options.retry_delay.as_millis(),
-                        operation_retry_options.max_retries, text.unwrap_or_default());
-
-                        current_retries += 1;
-                        client
-                            .executor
-                            .delay(operation_retry_options.retry_delay)
-                            .await;
-
-                        // we need to look up again the topic's address
-                        let prev = addr;
-                        addr = client.lookup_topic(&topic).await?;
-                        if prev != addr {
-                            info!(
-                                "topic {} moved: previous = {:?}, new = {:?}",
-                                topic, prev, addr
-                            );
+                Err(ConnectionError::PulsarError(Some(err), text)) => {
+                    if matches!(
+                        err,
+                        proto::ServerError::ServiceNotReady | proto::ServerError::ConsumerBusy
+                    ) {
+                        // Pulsar retryable error
+                        match operation_retry_options.max_retries {
+                            Some(max_retries) if current_retries < max_retries => {
```

Comment:
Previously this branch matches `None` also?

Reply:
The None can be considered as the variant, we have an max_retries but it was exausted.

But if you prefer, we can split between `None` and `Some(max_retries) if current_retries >= max_retries)`

Reply:
I read the previous condition guard this block is:

```rust
                    if operation_retry_options.max_retries.is_none()
                        || operation_retry_options.max_retries.unwrap() > current_retries
```

but now:

```rust
                        match operation_retry_options.max_retries {
                            Some(max_retries) if current_retries < max_retries => {
```

Reply:
The condition is the same, but now max-retries is not at the right position of the condition